packages/wonder-blocks-data/src/util/types.js

```diff
@@ -1,8 +1,15 @@
 // @flow
-export type ValidData = string | boolean | number | {...} | Array<ValidData>;
-
-export type Status = "loading" | "success" | "error";
+/**
+ * Define what can be cached.
+ *
+ * We disallow functions and undefined as undefined represents a cache miss
+ * and functions are not allowed.
+ */
```

Comment:
Wanted to add some commentary. In hindsight, I'd rename a bunch of these types but I didn't want to introduce more breaking changes.